src/components/ProtectedRoute/ProtectedRoute.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+function ProtectedRoute({ children, loggedIn, openLogin, ...props }) {
+  
+  useEffect(() => { if (!loggedIn) { openLogin(); } }, [])
```

Comment:
* If I'm authorized and I visit `saved articles` and then I reload the page (being on that page) the sign in popup opens (but I'm authorized and the token is fresh). The popup should be open only if I'm not authorized and try to access the `saved articles`. See here http://joxi.ru/gmvXnBgCeyLGLA